### ISSUE TITLE
Bucket handler always returns iron bucket

### ISSUE THREAD

**Issue by DarkGuardsman** [WayofTime/BloodMagic#420]
https://github.com/WayofTime/BloodMagic/blob/master/src/main/java/WayofTime/alchemicalWizardry/common/LifeBucketHandler.java#L18

It seems the fill bucket event is not checking to ensure the bucket being passed into the event is an iron bucket. Instead it just returns a filled bucket causes the player to get a free iron bucket out of a non-iron bucket.

**Comment by WayofTime** [comment]
Can you give an instance of this in-game?

**Comment by Arcaratus** [comment]
Probably when using Ex Nihilo etc.

**Comment by DarkGuardsman** [comment]
An example of this is a wooden bucket I coded up for a mod. In build 1 it would turned into an iron bucket of some of blood magic's fluid. Current build no longer has this issue as it seems most mods assume the ItemStack inputed into the FillBucketEvent is an iron bucket. So I have removed the forge event however it is still a bug that could effect other mods. 

Here is a video of what happens https://youtu.be/ilg5-Ln-l_0
